pkg/trace/api/api.go

```diff
@@ -227,20 +225,12 @@ func (r *HTTPReceiver) replyTraces(v Version, w http.ResponseWriter) {
 
 // handleTraces knows how to handle a bunch of traces
 func (r *HTTPReceiver) handleTraces(v Version, w http.ResponseWriter, req *http.Request) {
-	if atomic.LoadInt64(&r.refuse) == 1 {
-		// using too much memory
-		io.Copy(ioutil.Discard, req.Body)
-		w.WriteHeader(http.StatusTooManyRequests)
-		io.WriteString(w, fmt.Sprintf("request rejected; trace-agent is past memory threshold (apm_config.max_memory: %.0f bytes)", r.conf.MaxMemory))
-		metrics.Count("datadog.trace_agent.receiver.payload_refused", 1, []string{"reason:mem"}, 1)
-		return
-	}
 	if !r.PreSampler.Sample(req) {
 		// using too much CPU
 		io.Copy(ioutil.Discard, req.Body)
 		w.WriteHeader(http.StatusTooManyRequests)
-		io.WriteString(w, fmt.Sprintf("request rejected; trace-agent is past cpu threshold (apm_config.max_cpu_percent: %.1f)", r.conf.MaxCPU*100))
-		metrics.Count("datadog.trace_agent.receiver.payload_refused", 1, []string{"reason:cpu"}, 1)
+		io.WriteString(w, fmt.Sprintf("memory or CPU threshold exceeded; check trace-agent logs"))
+		metrics.Count("datadog.trace_agent.receiver.payload_refused", 1, nil, 1)
```

Comment:
Nit: we're loosing the "reason" of the refusal here. Not for this PR maybe, I'm OK with a follow-up, but maybe we could keep that level of detail by adding a `reason` arg to the `SetRate` method called here https://github.com/DataDog/datadog-agent/pull/3401/files#diff-f4f2947f405ab01c53d35418cdb59bebR399

I'll let you figure out, it's just an idea.

Reply:
Indeed, this is the only concern I had too. I tried to abstain myself from complicating the logic of the presampler by adding the reason and wanted to get reviewer's opinions too. It might be enough to have it only in the trace-agent logs. If people feel we should add it back, I'm happy to do so.

Reply:
Could add counter metrics down by the log lines `log.Warnf("CPU threshold exceeded (apm_config.max_cpu_percent` to track the number of times we've exceeded each threshold and that would give us that same information `datadog.trace_agent.presampler.exceeded, threshold:{threshold_name}`

Reply:
Good idea! Maybe something for #3403.